src/main/scala/edu/ie3/simona/agent/grid/GridAgentFailFast.scala

```diff
@@ -0,0 +1,81 @@
+/*
+ * © 2022. TU Dortmund University,
+ * Institute of Energy Systems, Energy Efficiency and Energy Economics,
+ * Research group Distribution grid planning and operation
+ */
+
+package edu.ie3.simona.agent.grid
+
+import edu.ie3.simona.agent.grid.GridAgentData.GridAgentInitData
+import edu.ie3.simona.config.SimonaConfig
+import edu.ie3.simona.exceptions.agent.GridAgentInitializationException
+
+import java.util.UUID
+import scala.jdk.CollectionConverters._
+
+case object GridAgentFailFast {
+
+  def failFast(
+      gridAgentInitData: GridAgentInitData,
+      simonaConfig: SimonaConfig
+  ): Unit = {
+
+    /** Check if there is InitData for superior or inferior GridGates
+      */
+    if (
+      gridAgentInitData.superiorGridGates.isEmpty && gridAgentInitData.inferiorGridGates.isEmpty
+    )
+      throw new GridAgentInitializationException(
+        s"${gridAgentInitData.subGridContainer.getGridName} has neither superior nor inferior grids! This can either " +
+          s"be cause by wrong subnetGate information or invalid parametrization of the simulation!"
+      )
+
+    /** Check if there exits voltage measurements for transformerControlGroups
+      */
+
+    val maybeControlConfig: Option[SimonaConfig.Simona.Control] =
+      simonaConfig.simona.control
+
+    val measurementUnits =
+      gridAgentInitData.subGridContainer.getRawGrid.getMeasurementUnits.asScala
+
+    val transformerUnits2W =
+      gridAgentInitData.subGridContainer.getRawGrid.getTransformer2Ws.asScala
+
+    val transformerUnits3W =
+      gridAgentInitData.subGridContainer.getRawGrid.getTransformer3Ws.asScala
+
+    if (maybeControlConfig.isDefined) {
+      maybeControlConfig.foreach(control =>
+        control.transformer.foreach(transformer =>
+          transformer.transformers.map(UUID.fromString).foreach {
+            transformers =>
+              // Check if transformer is part of subgrid of this GridAgent
+              val transformerUnit2W = transformerUnits2W
+                .find(element => element.getUuid == transformers)
+              val transformerUnit3W = transformerUnits3W
+                .find(element => element.getUuid == transformers)
+              if (transformerUnit2W.isDefined || transformerUnit3W.isDefined) {
+
+                transformer.measurements
+                  .map(UUID.fromString)
+                  .foreach { measurements =>
+                    val measurementUnit = measurementUnits
+                      .find(element => element.getUuid == measurements)
+                      .getOrElse(
+                        throw new GridAgentInitializationException(
```

Comment:
Currently, once at least one transformer belongs to the sub grid, all measurement units of the control group have to be present in the sub grid. Is this how we want it? I'm not quite sure myself. I think if a control group can be used on more than one sub grid, necessarily all the measurement units are present in only one of the sub grids. Thus, all measurement units can't be present in every sub grid.

Reply:
Yes thats my understanding as well. We might should change this when there will be controlGroups in future regulating at HV/MV transformer and measurements checking the influence at LV. But I'm not sure how realistic it will be that this will happen. Another case: 3W transformer with measurement in both subgrids. So far I would life with this, but let's discuss.

Reply:
I'm fine either way, I think we should just make it clear to the user if measurement units and transformers _have to_ sit in the same sub grid. Which brings me to the point of documentation... Maybe we should at least provide some very basic documentation in readthedocs here.

Reply:
Control groups are know part of documentation (basic version, might be extended later)